### ISSUE TITLE
`tf.keras.activations.swish` has wrong gradient

### ISSUE THREAD

**Issue by VictoriaGriffith** [tensorflow/tensorflow#57481]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

source

### Tensorflow Version

tf 2.9

### Custom Code

Yes

### OS Platform and Distribution

Linux Ubuntu 20.04

### Python version

3.9

### Current Behaviour?

In the example, `tf.keras.activations.swish` accepts a complex input, and `tf.test.compute_gradient` checks the theoretical and numerical gradient, and the two gradients are different. For example, on [0,1], the theoretical gradient is `0.56998741`, and the numerical gradient is `-0.56998753`.

Probably related to https://github.com/tensorflow/tensorflow/issues/57357

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
x = tf.complex(tf.random.uniform([1, 20, 1], maxval=0, dtype=tf.float64),tf.random.uniform([1, 20, 1], maxval=1.000000, dtype=tf.float64))

def f(x):
    return tf.keras.activations.swish(x, )
theoretical, numerical = tf.test.compute_gradient(f, [x])
print(theoretical)
print(numerical)

assert tf.experimental.numpy.allclose(theoretical, numerical, atol=1e-3, rtol=1e-3)
```

### Relevant log output

```shell
(array([[ 0.5       ,  0.56998741,  0.        , ...,  0.        ,
         0.        ,  0.        ],
       [-0.56998741,  0.5       ,  0.        , ...,  0.        ,
         0.        ,  0.        ],
       [ 0.        ,  0.        ,  0.5       , ...,  0.        ,
         0.        ,  0.        ],
       ...,
       [ 0.        ,  0.        ,  0.        , ...,  0.5       ,
         0.        ,  0.        ],
       [ 0.        ,  0.        ,  0.        , ...,  0.        ,
         0.5       ,  0.1744664 ],
       [ 0.        ,  0.        ,  0.        , ...,  0.        ,
        -0.1744664 ,  0.5       ]]),)
(array([[ 0.5       , -0.56998753,  0.        , ...,  0.        ,
         0.        ,  0.        ],
       [ 0.56998728,  0.5       ,  0.        , ...,  0.        ,
         0.        ,  0.        ],
       [ 0.        ,  0.        ,  0.5       , ...,  0.        ,
         0.        ,  0.        ],
       ...,
       [ 0.        ,  0.        ,  0.        , ...,  0.5       ,
         0.        ,  0.        ],
       [ 0.        ,  0.        ,  0.        , ...,  0.        ,
         0.5       , -0.17446643],
       [ 0.        ,  0.        ,  0.        , ...,  0.        ,
         0.17446637,  0.5       ]]),)
AssertionError
```
</details>

**Comment by sushreebarsa** [comment]
@VictoriaGriffith I was able to replicate the issue on colab, please find the [gist](https://colab.research.google.com/gist/sushreebarsa/f644fe28f2a9b87ca7a4029ea9fcf210/57481.ipynb) here.
This issue seems to be Keras issue.  Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues) as Keras development is fully moving to [github.com/keras-team/keras](http://github.com/keras-team/keras). All issues and PRs related to keras will be addressed in that repo.
To know more see this TF forum discussion ; 
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999)
Thank you!

**Comment by VictoriaGriffith** [comment]
Thanks @sushreebarsa for replicating the bug.
I have reported this issue in Keras repo https://github.com/keras-team/keras/issues/16962. 
However, I would still like to keep it here, because according to https://github.com/tensorflow/tensorflow/issues/57357, this bug seems to come from the same gradient problem as `tf.nn.silu` and `tf.nn.sigmoid` (because Swish computes `x*sigmoid(x)`), and could be related to tensorflow implementation for this sigmoid gradient computation.

**Comment by cantonios** [comment]
Again, looks simply to be a numerical issue with estimating the gradients.  Playing around with `delta` for the numerical causes significant swings in the maximum error.  The theoretical gradient of `x * sigmoid(x)` is correct from #57357.
